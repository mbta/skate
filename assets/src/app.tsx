// We need to import the CSS so that webpack will load it.
// The MiniCssExtractPlugin is used to separate it out into
// its own CSS file.
declare function require(name: string): string
// tslint:disable-next-line
require("../css/app.scss")

// webpack automatically bundles all modules in your
// entry points. Those entry points can be configured
// in "webpack.config.js".
//
// Import dependencies
//
import "@babel/polyfill"
import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.webpack.css" // see https://github.com/Leaflet/Leaflet/issues/4968#issuecomment-483402699
import "leaflet/dist/leaflet.css"
import "phoenix_html"
import * as React from "react"
import ReactDOM from "react-dom"
import AppStateWrapper from "./components/appStateWrapper"

if (window.FS && window.username) {
  window.FS.identify(window.username, { displayName: window.username })
}

// Import local files
//
// Local files can be imported directly using relative paths, for example:
// import socket from "./socket"

ReactDOM.render(<AppStateWrapper />, document.getElementById("app"))
