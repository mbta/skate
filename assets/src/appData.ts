const appData = (): DOMStringMap | undefined => {
  const dataEl = document.getElementById("app")

  if (!dataEl) {
    return undefined
  }

  return dataEl.dataset
}

export default appData
