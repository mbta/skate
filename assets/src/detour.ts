import { LatLngLiteral } from "leaflet"
import { Shape, ShapePoint } from "./schedule"

export interface DetourShape {
  coordinates: ShapePoint[]
  directions: {
    instruction: string
  }[]
}

export interface OriginalRoute {
  routeName: string
  routeDescription: string
  routeOrigin: string
  routeDirection: string
  shape: Shape
  center: LatLngLiteral
  zoom: number
}
