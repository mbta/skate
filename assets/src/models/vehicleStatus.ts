import featureIsEnabled from "../laboratoryFeatures"
import { Vehicle } from "../realtime.d"

/** Where a vehicle is relative to its schedule.
 * Swiftly assigns this to all vehicles.
 */
export type OnTimeStatus = "on-time" | "early" | "late"

export enum HeadwaySpacing {
  VeryBunched = "very-bunched",
  Bunched = "bunched",
  Ok = "ok",
  Gapped = "gapped",
  VeryGapped = "very-gapped",
}

/** How the vehicle triangle should be drawn,
 * taking into account the vehicle's state and the settings
 */
export type DrawnStatus = OnTimeStatus | "off-course" | "ghost" | "plain"

export const onTimeStatus = (scheduleAdherenceSecs: number): OnTimeStatus => {
  const oneMinuteInSeconds = 60
  const sixMinutesInSeconds = 360

  if (scheduleAdherenceSecs < -oneMinuteInSeconds) {
    return "early"
  } else if (scheduleAdherenceSecs > sixMinutesInSeconds) {
    return "late"
  } else {
    return "on-time"
  }
}

export const drawnStatus = (vehicle: Vehicle): DrawnStatus => {
  if (vehicle.isOffCourse) {
    return "off-course"
  } else if (
    isShuttle(vehicle) ||
    (featureIsEnabled("headway_ladder_colors") &&
      vehicle.headwaySpacing !== null)
  ) {
    // Headway lines give the status instead of the vehicles
    return "plain"
  } else {
    return onTimeStatus(vehicle.scheduleAdherenceSecs)
  }
}

export const humanReadableScheduleAdherence = (vehicle: Vehicle): string =>
  vehicle.isOffCourse
    ? "Invalid"
    : humanReadableOnTimeStatus(onTimeStatus(vehicle.scheduleAdherenceSecs))

const humanReadableOnTimeStatus = (status: OnTimeStatus): string => {
  switch (status) {
    case "early":
      return "early"

    case "on-time":
      return "on time"

    case "late":
      return "late"
  }
}

export const humanReadableHeadwaySpacing = (
  spacing: HeadwaySpacing | null
): string => {
  switch (spacing) {
    case null:
      return "good"

    case HeadwaySpacing.VeryGapped:
      return "very gapped"

    case HeadwaySpacing.Gapped:
      return "gapped"

    case HeadwaySpacing.Ok:
      return "good"

    case HeadwaySpacing.Bunched:
      return "bunched"

    case HeadwaySpacing.VeryBunched:
      return "very bunched"
  }
}

export const headwaySpacingToString = (spacing: HeadwaySpacing): string => {
  switch (spacing) {
    case HeadwaySpacing.VeryGapped:
      return "very-gapped"

    case HeadwaySpacing.Gapped:
      return "gapped"

    case HeadwaySpacing.Ok:
      return "ok"

    case HeadwaySpacing.Bunched:
      return "bunched"

    case HeadwaySpacing.VeryBunched:
      return "very-bunched"
  }
}

export const statusClass = (status: DrawnStatus): string => {
  switch (status) {
    case "plain":
      return ""

    default:
      return status
  }
}

const isShuttle = (vehicle: Vehicle): boolean =>
  (vehicle.runId || "").startsWith("999")
