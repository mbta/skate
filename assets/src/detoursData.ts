import { array, Infer, number, string, type } from "superstruct"
import { DraftDetours } from "./models/detours"

export const DraftDetoursData = array(
  type({
    route: string(),
    direction: string(),
    name: string(),
    intersection: string(),
    active_since: number(),
  })
)
export type DraftDetoursData = Infer<typeof DraftDetoursData>

export const draftDetoursFromData = (
  draftDetoursData: DraftDetoursData
): DraftDetours =>
  draftDetoursData.map((draft) => ({
    route: draft.route,
    direction: draft.direction,
    name: draft.name,
    intersection: draft.intersection,
    activeSince: draft.active_since,
  }))
