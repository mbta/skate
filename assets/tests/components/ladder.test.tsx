import { mount } from "enzyme"
import React from "react"
import renderer from "react-test-renderer"
import Ladder, { LadderDirection } from "../../src/components/ladder"
import { HeadwaySpacing } from "../../src/models/vehicleStatus"
import DispatchProvider from "../../src/providers/dispatchProvider"
import { Vehicle } from "../../src/realtime.d"
import { TimepointId } from "../../src/schedule.d"
import { selectVehicle } from "../../src/state"

jest.mock("../../src/laboratoryFeatures", () => ({
  __esModule: true,
  default: () => true,
}))

describe("ladder", () => {
  test("renders a ladder", () => {
    const timepoints = ["t0", "t1", "t2"]
    const vehicles: Vehicle[] = [
      {
        id: "upward",
        label: "upward",
        runId: "run-1",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 0,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op1",
        operatorName: "SMITH",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.5,
          timepointId: "t1",
        },
        scheduledLocation: null,
      },
      {
        id: "downward",
        label: "downward",
        runId: "run-2",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 1,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op2",
        operatorName: "JONES",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.75,
          timepointId: "t2",
        },
        scheduledLocation: {
          directionId: 0,
          timepointStatus: {
            timepointId: "t2",
            fractionUntilTimepoint: 0.75,
          },
        },
      },
      {
        id: "notimepoint",
        label: "notimepoint",
        runId: "run-3",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 1,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op3",
        operatorName: "XI",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: null,
        scheduledLocation: null,
      },
    ]
    const ladderDirection = LadderDirection.ZeroToOne

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("filters out vehicles whose block is not active", () => {
    const timepoints = ["t0", "t1", "t2"]
    const vehicles: Vehicle[] = [
      {
        id: "upward",
        label: "upward",
        runId: "run-1",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 0,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op1",
        operatorName: "SMITH",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: false,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.5,
          timepointId: "t1",
        },
        scheduledLocation: null,
      },
      {
        id: "downward",
        label: "downward",
        runId: "run-2",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 1,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op2",
        operatorName: "JONES",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.75,
          timepointId: "t2",
        },
        scheduledLocation: {
          directionId: 0,
          timepointStatus: {
            timepointId: "t2",
            fractionUntilTimepoint: 0.75,
          },
        },
      },
    ]
    const ladderDirection = LadderDirection.ZeroToOne

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("shows schedule line in the other direction", () => {
    const timepoints = ["t0", "t1", "t2"]
    const vehicles: Vehicle[] = [
      {
        id: "downward",
        label: "downward",
        runId: "run-2",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 0,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op2",
        operatorName: "JONES",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.75,
          timepointId: "t2",
        },
        scheduledLocation: {
          directionId: 1,
          timepointStatus: {
            timepointId: "t2",
            fractionUntilTimepoint: 0.75,
          },
        },
      },
    ]
    const ladderDirection = LadderDirection.OneToZero
    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("highlights a selected vehicle", () => {
    const timepoints = ["t0", "t1", "t2"]
    const vehicles: Vehicle[] = [
      {
        id: "upward",
        label: "upward",
        runId: "run-1",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 0,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op1",
        operatorName: "SMITH",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.5,
          timepointId: "t1",
        },
        scheduledLocation: null,
      },
      {
        id: "downward",
        label: "downward",
        runId: "run-2",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 1,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op2",
        operatorName: "JONES",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.75,
          timepointId: "t2",
        },
        scheduledLocation: {
          directionId: 0,
          timepointStatus: {
            timepointId: "t2",
            fractionUntilTimepoint: 0.75,
          },
        },
      },
    ]
    const ladderDirection = LadderDirection.ZeroToOne

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={"upward"}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("clicking a vehicle selects that vehicle", () => {
    const mockDispatch = jest.fn()

    const timepoints = ["t0", "t1", "t2"]
    const vehicle: Vehicle = {
      id: "upward",
      label: "upward",
      runId: "run-1",
      timestamp: 0,
      latitude: 0,
      longitude: 0,
      directionId: 0,
      routeId: "route",
      tripId: "trip",
      headsign: null,
      viaVariant: null,
      operatorId: "op1",
      operatorName: "SMITH",
      bearing: 33,
      speed: 50.0,
      blockId: "block-1",
      headwaySecs: 859.1,
      headwaySpacing: HeadwaySpacing.Ok,
      previousVehicleId: "v2",
      scheduleAdherenceSecs: 0,
      scheduleAdherenceString: "0.0 sec (ontime)",
      scheduleAdherenceStatus: "on-time",
      scheduledHeadwaySecs: 120,
      isOffCourse: false,
      blockIsActive: true,
      dataDiscrepancies: [],
      stopStatus: {
        status: "in_transit_to",
        stopId: "stop",
        stopName: "stop",
      },
      timepointStatus: {
        fractionUntilTimepoint: 0.5,
        timepointId: "t1",
      },
      scheduledLocation: null,
    }

    const ladderDirection = LadderDirection.ZeroToOne

    const wrapper = mount(
      <DispatchProvider dispatch={mockDispatch}>
        <Ladder
          timepoints={timepoints}
          vehicles={[vehicle]}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      </DispatchProvider>
    )
    wrapper.find(".m-ladder__vehicle").simulate("click")

    expect(mockDispatch).toHaveBeenCalledWith(selectVehicle(vehicle.id))
  })

  test("renders a ladder with no timepoints", () => {
    const timepoints: TimepointId[] = []
    const vehicles: Vehicle[] = [
      {
        id: "upward",
        label: "upward",
        runId: "run-1",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 0,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op1",
        operatorName: "SMITH",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.5,
          timepointId: "t1",
        },
        scheduledLocation: null,
      },
    ]
    const ladderDirection = LadderDirection.ZeroToOne

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("renders a reversed ladder", () => {
    const timepoints = ["t0", "t1", "t2"]
    const vehicles: Vehicle[] = [
      {
        id: "upward",
        label: "upward",
        runId: "run-1",
        timestamp: 0,
        latitude: 0,
        longitude: 0,
        directionId: 1,
        routeId: "route",
        tripId: "trip",
        headsign: null,
        viaVariant: null,
        operatorId: "op1",
        operatorName: "SMITH",
        bearing: 33,
        speed: 50.0,
        blockId: "block-1",
        headwaySecs: 859.1,
        headwaySpacing: HeadwaySpacing.Ok,
        previousVehicleId: "v2",
        scheduleAdherenceSecs: 0,
        scheduleAdherenceString: "0.0 sec (ontime)",
        scheduleAdherenceStatus: "on-time",
        scheduledHeadwaySecs: 120,
        isOffCourse: false,
        blockIsActive: true,
        dataDiscrepancies: [],
        stopStatus: {
          status: "in_transit_to",
          stopId: "stop",
          stopName: "stop",
        },
        timepointStatus: {
          fractionUntilTimepoint: 0.5,
          timepointId: "t1",
        },
        scheduledLocation: {
          directionId: 1,
          timepointStatus: {
            timepointId: "t1",
            fractionUntilTimepoint: 0.4,
          },
        },
      },
    ]
    const ladderDirection = LadderDirection.OneToZero

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={vehicles}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test("renders an off-course vehicle", () => {
    const vehicle: Vehicle = {
      id: "y1439",
      label: "1439",
      runId: "run-1",
      timestamp: 1562777122,
      latitude: 42.38954,
      longitude: -71.07405,
      directionId: 0,
      routeId: "71",
      tripId: "40725309",
      headsign: "Harvard",
      viaVariant: "D",
      operatorId: "op1",
      operatorName: "SMITH",
      bearing: 0,
      speed: 0,
      blockId: "T71-17",
      headwaySecs: 859.1,
      headwaySpacing: HeadwaySpacing.Ok,
      previousVehicleId: "v2",
      scheduleAdherenceSecs: 0,
      scheduleAdherenceString: "0.0 sec (ontime)",
      scheduleAdherenceStatus: "on-time",
      scheduledHeadwaySecs: 120,
      isOffCourse: true,
      blockIsActive: true,
      dataDiscrepancies: [
        {
          attribute: "trip_id",
          sources: [
            {
              id: "busloc",
              value: "40725309",
            },
            {
              id: "swiftly",
              value: null,
            },
          ],
        },
      ],
      stopStatus: {
        status: "in_transit_to",
        stopId: "stop",
        stopName: "stop",
      },
      timepointStatus: null,
      scheduledLocation: {
        directionId: 0,
        timepointStatus: {
          timepointId: "mtsty",
          fractionUntilTimepoint: 0,
        },
      },
    }

    const timepoints = ["t0", "t1", "t2"]
    const ladderDirection = LadderDirection.ZeroToOne

    const tree = renderer
      .create(
        <Ladder
          timepoints={timepoints}
          vehicles={[vehicle]}
          ladderDirection={ladderDirection}
          selectedVehicleId={undefined}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
