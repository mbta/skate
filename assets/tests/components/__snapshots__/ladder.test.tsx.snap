// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ladder highlights a selected vehicle 1`] = `
<div
  className="m-ladder"
>
  <svg
    className="m-ladder__svg"
    height={500}
    viewBox="-92 -40 184 500"
    width={184}
  >
    0
    <g>
      <line
        className="m-ladder__scheduled-line on-time"
        x1={-63}
        x2={-40}
        y1={262.5}
        y2={262.5}
      />
    </g>
    <g>
      <g
        className="m-ladder__vehicle on-time selected"
        onClick={[Function]}
        transform="translate(63,315)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={11.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={17}
          >
            upward
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(0) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <g>
      <g
        className="m-ladder__vehicle on-time "
        onClick={[Function]}
        transform="translate(-63,262.5)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={-22.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={-17}
          >
            downward
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(180) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <line
      className="m-ladder__line"
      x1={-40}
      x2={-40}
      y1="0"
      y2={420}
    />
    <line
      className="m-ladder__line"
      x1={40}
      x2={40}
      y1="0"
      y2={420}
    />
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={0}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={0}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={0}
    >
      t2
    </text>
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={210}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={210}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={210}
    >
      t1
    </text>
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={420}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={420}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={420}
    >
      t0
    </text>
  </svg>
</div>
`;

exports[`ladder renders a ladder 1`] = `
<div
  className="m-ladder"
>
  <svg
    className="m-ladder__svg"
    height={500}
    viewBox="-92 -40 184 500"
    width={184}
  >
    0
    0
    <g>
      <line
        className="m-ladder__scheduled-line on-time"
        x1={-63}
        x2={-40}
        y1={262.5}
        y2={262.5}
      />
    </g>
    <g>
      <g
        className="m-ladder__vehicle on-time "
        onClick={[Function]}
        transform="translate(63,315)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={11.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={17}
          >
            upward
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(0) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <g>
      <g
        className="m-ladder__vehicle on-time "
        onClick={[Function]}
        transform="translate(-63,0)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={-22.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={-17}
          >
            notimepoint
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(180) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <g>
      <g
        className="m-ladder__vehicle on-time "
        onClick={[Function]}
        transform="translate(-63,262.5)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={-22.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={-17}
          >
            downward
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(180) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <line
      className="m-ladder__line"
      x1={-40}
      x2={-40}
      y1="0"
      y2={420}
    />
    <line
      className="m-ladder__line"
      x1={40}
      x2={40}
      y1="0"
      y2={420}
    />
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={0}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={0}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={0}
    >
      t2
    </text>
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={210}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={210}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={210}
    >
      t1
    </text>
    <circle
      className="m-ladder__stop-circle"
      cx={-40}
      cy={420}
      r="3"
    />
    <circle
      className="m-ladder__stop-circle"
      cx={40}
      cy={420}
      r="3"
    />
    <text
      className="m-ladder__timepoint-name"
      dominantBaseline="middle"
      textAnchor="middle"
      x="0"
      y={420}
    >
      t0
    </text>
  </svg>
</div>
`;

exports[`ladder renders a ladder with no timepoints 1`] = `
<div
  className="m-ladder"
>
  <svg
    className="m-ladder__svg"
    height={500}
    viewBox="-92 -40 184 500"
    width={184}
  >
    0
    <g>
      <g
        className="m-ladder__vehicle on-time "
        onClick={[Function]}
        transform="translate(63,0)"
      >
        <g
          className="m-vehicle-icon m-vehicle-icon--medium"
        >
          <rect
            className="m-vehicle-icon__label-background"
            height={11}
            rx={5.5}
            ry={5.5}
            width={26}
            x={-13}
            y={11.5}
          />
          <text
            className="m-vehicle-icon__label"
            dominantBaseline="central"
            textAnchor="middle"
            x="0"
            y={17}
          >
            upward
          </text>
          <path
            className="m-vehicle-icon__triangle"
            d="m27.34 9.46 16.84 24.54a4.06 4.06 0 0 1 -1 5.64 4.11 4.11 0 0 1 -2.3.71h-33.72a4.06 4.06 0 0 1 -4.06-4.11 4 4 0 0 1 .72-2.24l16.84-24.54a4.05 4.05 0 0 1 5.64-1.05 4 4 0 0 1 1.04 1.05z"
            transform="scale(0.625) rotate(0) translate(-24,-22)"
          />
        </g>
      </g>
    </g>
    <line
      className="m-ladder__line"
      x1={-40}
      x2={-40}
      y1="0"
      y2={420}
    />
    <line
      className="m-ladder__line"
      x1={40}
      x2={40}
      y1="0"
      y2={420}
    />
  </svg>
</div>
`;
